### DISCUSSION TITLE
BBN Forums

### DISCUSSION THREAD

**Discussion by aviljoenn** [bareboat-necessities/lysmarine_gen#137] [General]
Hi All,

Which forum would be the best and most active for BBN Marine Linux OS discussions?
E.g. do the general discussions and questions happen here or Openmarine or elsewhere?
Thank you so much and looking forward to some interaction.
Kind Regards
Andre

**Comment by mgrouch** [comment]
@aviljoenn 
Here would be the best place.

**Comment by robmicro** [comment]
WOW - stunning project! I'm not so good in raspberry, yet. But wondered if it is possible to install BBN Marine on a desktop pc or mac? E.g. via a Raspberry Pi Desktop image package? I'd prefer to use an old laptop for navigation instead of a raspy... Or can you tell me how to do it myself?
Thanks,
Rob

**Reply by mgrouch** [reply]
It’s not possible. The current implementation is for raspberry pi4, CM4, or pi400 only.

**Reply by robmicro** [reply]
Even not with raspios on a desktop and a local git clone + In place build -> install.sh?

**lysmarine_gen readme.mdt says...**

At this time, The tested SBCs are :
    RaspberryPi 3 (armhf and arm64)
    PineA64 LTS (based on the Pine64so kernel)
    Liveusb
    **Desktop install**

**Reply by mgrouch** [reply]
You might be reading it on 
https://github.com/lysmarine/lysmarine_gen

It’s not BBN OS. BBN OS is for raspberry pi4/pi400/CM4

**Reply by robmicro** [reply]
I know - just thought because BBN OS is a fork from lsysmarine - or am I wrong - sorry about!

> This is the fork of the original LysMarine https://github.com/lysmarine/lysmarine_gen by Frederic Guilbault. It is based on the LysMarine OS, but differs from it in a number of included applications, and the UI features. Now it is a distinct OS image.

**Reply by mgrouch** [reply]
Even if you try building locally it will not build for unsupported platforms without significant changes.

**Reply by robmicro** [reply]
ah, ok. I assume there’s no intention for the future to make this possible?

**Reply by mgrouch** [reply]
Not sure who would needs it. 
pi is lower power usage than Intel. 
It has more interfaces i2c, 1w and so on. There wild variety of CM4 boards from MCUZone, Waveshare, etc.
There is no point to use Intel on a recreational boat.

**Reply by robmicro** [reply]
that’s simple. I have e.g. some old macbooks laying around and on my boat, I have NMEA in LAN or Wifi available. Instead buying a screen and a pi and mount everything somewhere, I thought to use a macbook instead… can move it around and screen is available, too. It’s just convenient to have marine related apps preconfigured in a project like this, saves a lot of time on setting up and service it over time. especially if you want to test different apps, too.

**Reply by mgrouch** [reply]
I don’t even have MacBook to build and test it. If you wish attempting
it yourself you can start with Debian (I guess Buster minimal image for Mac) 
and work on changing scripts for it. 
You will throw away some raspberry specific steps.

**Reply by robmicro** [reply]
I’ll try and play around with this here: https://www.raspberrypi.com/software/raspberry-pi-desktop/
Which scripts should I look at first?

**Reply by mgrouch** [reply]
This is the starting point
https://github.com/bareboat-necessities/my-bareboat/blob/master/bbn-os-build/bbn-os-build-on-ubuntu.sh

**Comment by mbohmier** [comment]
I installed the new version of Lysmarine  (2022 07 18) and got it running.
I got the internet working and the VNC. Now I have to figure out how to start the GPS and pypilot motor controller.

**Reply by mgrouch** [reply]
It’s all in the docs. GPS is almost plug and play. Here are the steps for motor controller 

https://bareboat-necessities.github.io/my-bareboat/bareboat-os.html#_pypilot
